system.go

```diff
@@ -11,12 +11,24 @@ import (
 )
 
 func countOpenFiles() int {
-	out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %v", os.Getpid())).Output()
-	if err != nil {
-		log.Fatal(err)
+
+	if runtime.GOOS == "linux" {
+
+		if isCommandAvailable("lsof") {
+			out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("lsof -p %d", os.Getpid())).Output()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			lines := strings.Split(string(out), "\n")
+			return len(lines) - 1
+		}
+
+		log.Fatalln("lsof command does not exist.kindly run sudo-apt-get install lsof")
+
 	}
-	lines := strings.Split(string(out), "\n")
-	return len(lines) - 1
+	log.Fatalln("this command is not available on non-linux based operating systems")
```

Comment:
So Glutton doesn't run on any non Linux OS?

Reply:
You said glutton was only for Linux :)

Reply:
But then what's the point of `runtime.GOOS == "linux"` ?

Reply:
Yeah I agree with you we could then make the message "command not available" regardless.?

Reply:
How about let the function return an error and not report the number of open files on error?

Reply:
I do not think that would help much, we could rather do a `log.Fatal` which would anyways cause a `sys.exit()` and would exit the function and if we even return an error we would by default have to return some value for int, which in that case would be 0.